PowerPointLabs/Test/FunctionalTest/PasteLabTest.cs

```diff
@@ -180,15 +187,46 @@ private void IsClipboardRestoredPasteIntoGroup(int originalSlideNo, int expSlide
             }, originalSlideNo, ShapeToCopyPrefix, expSlideNo, "", ShapeToCompareCopied);
         }
 
-        private void AssertIsSame(int actualSlideNo, int expectedSlideNo)
+        private void IsDegradedAfterReplaceWithClipboard(int originalSlideNo, int replacedSlideNo, int degradedSlideNo)
+        {
+            Microsoft.Office.Interop.PowerPoint.ShapeRange shapes = GetShapesByPrefix(replacedSlideNo, ShapeToCopyPrefix);
+            shapes.Cut();
+
+            PpOperations.SelectShapes(new List<string> { ShapeToReplace });
+            PplFeatures.ReplaceWithClipboard();
+
+            // Degradation results in subtle differences, therefore comparison threshold must be stricter
+            AssertIsSame(originalSlideNo, replacedSlideNo, 0.999999999);
+            
+            // Verify that the degradation is detectable
+            AssertNotSame(originalSlideNo, degradedSlideNo, 0.999999999);
+        }
+
+        private void AssertIsSame(int actualSlideNo, int expectedSlideNo, double similarityTolerance = 0.95)
         {
             Microsoft.Office.Interop.PowerPoint.Slide actualSlide = PpOperations.SelectSlide(actualSlideNo);
             Microsoft.Office.Interop.PowerPoint.Slide expectedSlide = PpOperations.SelectSlide(expectedSlideNo);
 
-            SlideUtil.IsSameLooking(expectedSlide, actualSlide);
+            SlideUtil.IsSameLooking(expectedSlide, actualSlide, similarityTolerance);
             SlideUtil.IsSameAnimations(expectedSlide, actualSlide);
         }
 
+        private void AssertNotSame(int actualSlideNo, int expectedSlideNo, double similarityTolerance = 0.95)
+        {
+            try
+            {
+                AssertIsSame(actualSlideNo, expectedSlideNo, similarityTolerance);
+                throw new AssertFailedException("AssertNotSame failed. The slides look similar.");
+            }
+            catch (AssertFailedException e)
+            {
+                if (e.Message.Equals("AssertNotSame failed. The slides look similar."))
+                {
+                    throw e;
+                }
+            }
+        }
+
```

Comment:
I am aware that this is not the most elegant way to implement this method, but I wanted to avoid having to introduce lots of duplicate code for a method that is essentially the opposite of `AssertIsSame`. Any suggested improvements you guys have would be well appreciated.

Reply:
A possible slightly-more-elegant-but-still-kinda-not-really-that-elegant way to do it can be as such:

```csharp
try
{
    AssertIsSame(actualSlideNo, expectedSlideNo, similarityTolerance);
}
catch (AssertFailedException e)
{
    return;
}

throw new AssertFailedException("AssertNotSame failed. The slides look similar.");
```

Reply:
Aiight thanks! This seems much cleaner